### ISSUE TITLE
Investigate Selenium 3 and geckodriver compatibility

### ISSUE THREAD

**Issue by gunesacar** [citp/OpenWPM#93]
Mozilla is already phasing out support for Selenium 2. It may be a good idea to start working towards Selenium 3 and geckodriver compatibility.

Here's a WIP branch, feel free to take it from there:
https://github.com/gunesacar/OpenWPM/tree/selenium3
https://travis-ci.org/gunesacar/OpenWPM/builds/162494675

I listed the remaining problems in the commit message and the code comments.
Most pressing ones are:
- No self-signed certificate support (https://bugzil.la/1103196)
- `send_keys` command doesn't work (hence the `tab_restart_browser`)

**Comment by englehardt** [comment]
Self-signed certificates aren't an issue anymore as we'll be removing the proxy in future releases.

**Comment by englehardt** [comment]
Selenium only officially supports FF 47 and below (though 45 is recommended as that's the latest ESR). Their new issue template includes the following:

```
Any issue logged here for Firefox 48 or later will be closed as a duplicate of
#2559. Our recommendation is to switch to GeckoDriver, or to continue testing
on Firefox 45 until GeckoDriver is a viable option for you.
```

Combing through some Selenium issues it seems like `FirefoxDriver` will continue to work with FF 48 and 49 as long as we use unbranded builds that don't require add-on signing (see: https://github.com/SeleniumHQ/selenium/issues/2942), so switching to unbranded builds could be a stop-gap until we can get geckodriver supported.

**Comment by alexristich** [comment]
Hi! Just checking to see when this might be planned. I followed up on the workaround noted in https://github.com/SeleniumHQ/selenium/issues/1181, but didn't receive a response. Looking back at this again and hoping something might not be too far off. Also, suggestions on how to implement the workaround are also most welcome!

**Comment by englehardt** [comment]
We likely won't make any headway on this until March / April when the next Firefox ESR is released. Aside from the engineering work required for OpenWPM, we are blocked on `send_keys` support, which seems to currently be supported in `GeckoDriver` but is not yet available in Selenium (See: https://github.com/SeleniumHQ/selenium/issues/2611).

**Comment by englehardt** [comment]
Added in #152
